src/lib/autoinstall/xml_checks.rb

```diff
@@ -31,35 +32,46 @@ module Y2Autoinstallation
   # for not well formed or invalid documents. It is possible to continue
   # anyway and ignore the found problems at your risk.
   class XmlChecks
-    extend Yast::Logger
-    extend Yast::I18n
-    textdomain "autoinst"
+    include Singleton
+    include Yast::Logger
+    include Yast::I18n
 
     # paths to the default AutoYaST schema files
     PROFILE_SCHEMA = "/usr/share/YaST2/schema/autoyast/rng/profile.rng".freeze
     RULES_SCHEMA = "/usr/share/YaST2/schema/autoyast/rng/rules.rng".freeze
     CLASSES_SCHEMA = "/usr/share/YaST2/schema/autoyast/rng/classes-use.rng".freeze
 
+    ERRORS_PATH = "/var/lib/YaST2/xml_checks_errors".freeze
+    # @return [Array<String>] reported errors md5
```

Comment:
I would say that md5 is implementation detail. I would say just hash or id of reported errors.